src/main/java/net/neoforged/neoforge/common/extensions/IBlockExtension.java

```diff
@@ -976,4 +977,23 @@ default PushReaction getPistonPushReaction(BlockState state) {
     default boolean isEmpty(BlockState state) {
         return state.is(Blocks.AIR) || state.is(Blocks.CAVE_AIR) || state.is(Blocks.VOID_AIR);
     }
+
+    /**
+     * Determines if this block can spawn Bubble Columns and if so, what direction the column flows.
+     * <p></p>
+     * NOTE: The block itself will still need to call {@link net.minecraft.world.level.block.BubbleColumnBlock#updateColumn(LevelAccessor, BlockPos, BlockState)} in their tick method and schedule a block tick in the block's onPlace.
+     * Also, schedule a fluid tick in updateShape method if update direction is up. Both are needed in order to get the Bubble Columns to function properly. See {@link net.minecraft.world.level.block.SoulSandBlock} and {@link net.minecraft.world.level.block.MagmaBlock} for example.
+     *
+     * @param state The current state
+     * @return BubbleColumnDirection.NONE for no Bubble Column. Otherwise, will spawn Bubble Column flowing with specified direction
+     */
+    default BubbleColumnDirection sustainBubbleColumn(BlockState state) {
```

Comment:
I'm not really a fan of this method name, but I can't think of anything better at the moment either

Reply:
allowedBubbleColumnDirection? Then none would make sense for no direction and thus, no column. Thoughts?

Reply:
`getAllowedBubbleColumnDirection()` would be my preference in that case, though "allowed" also sounds weird to me.
Maybe it should just be `getBubbleColumnDirection()` :thinking: